### PR TITLE
Users without valid birthdates were falsely selected

### PR DESCRIPTION
**Pull request by live627**
- https://www.simplemachines.org/community/index.php?msg=4103960
-  #7140

### CODE REVIEW

File: Sources/Subs-Calendar.php [hunk 1]
```diff
@@ -36,52 +36,38 @@ function getBirthdayRange($low_date, $high_date)
...
-			WHERE MONTH(birthdate) != {int:no_month}
-				AND DAYOFMONTH(birthdate) != {int:no_day}
-				AND YEAR(birthdate) <= {int:max_year}
+			WHERE birthdate != \'1004-01-01\'
 				AND (
 					DATE_FORMAT(birthdate, {string:year_low}) BETWEEN {date:low_date} AND {date:high_date}' . ($year_low == $year_high ? '' : '
 					OR DATE_FORMAT(birthdate, {string:year_high}) BETWEEN {date:low_date} AND {date:high_date}') . '
 				)
-				AND is_activated = {int:is_activated}',
+				AND is_activated = 1',
```
Comment (albertlast) [inline Sources/Subs-Calendar.php]: i was believing that smf coding style is,
that static value also should be placed by placeholder?

File: Sources/Subs-Calendar.php [hunk 2]
```diff
@@ -36,52 +36,38 @@ function getBirthdayRange($low_date, $high_date)
...
 	{
 		// Collect all of the birthdays for this month.  I know, it's a painful query.
-		$result = $smcFunc['db_query']('birthday_array', '
+		$result = $smcFunc['db_query']('', '
 			SELECT id_member, real_name, YEAR(birthdate) AS birth_year, birthdate
 			FROM {db_prefix}members
-			WHERE MONTH(birthdate) != {int:no_month}
-				AND DAYOFMONTH(birthdate) != {int:no_day}
-				AND YEAR(birthdate) <= {int:max_year}
+			WHERE birthdate != \'1004-01-01\'
```
Comment (albertlast) [inline Sources/Subs-Calendar.php]: please use date placeholder here ``{date:``

File: Sources/Subs-Calendar.php [hunk 3]
```diff
@@ -36,52 +36,38 @@ function getBirthdayRange($low_date, $high_date)
...
 	else
 	{
-		$result = $smcFunc['db_query']('birthday_array', '
+		$result = $smcFunc['db_query']('', '
 			SELECT id_member, real_name, YEAR(birthdate) AS birth_year, birthdate
 			FROM {db_prefix}members
-			WHERE YEAR(birthdate) != {string:year_one}
-				AND MONTH(birthdate) != {int:no_month}
-				AND DAYOFMONTH(birthdate) != {int:no_day}
+			WHERE birthdate != \'1004-01-01\'
```
Comment (albertlast) [inline Sources/Subs-Calendar.php]: please use date placeholder here ``{date:``

**Review by albertlast — CHANGES_REQUESTED** [review]
one general question
and please use the date placeholder.

**Comment by jdarwood007** [comment]
@live627 I agree with @albertlast here.  Can we use the proper place holders for the query?

The is_activated seems redundant now, but in future SMF versions, I would hope we can switch to using constants, so its cleaner code.

**Review by jdarwood007 — APPROVED** [review]

**Review by albertlast — APPROVED** [review]

**Pull request merged**